### PR TITLE
freecad: fix musl build

### PR DESCRIPTION
**Pull request by Piraty**
sorry, forgot to actually fix musl build issues when i removed the `broken` option in #13404.
It now compiles on x86_64-musl, but i don't run a desktop on that arch, so i didn't test the actual behviour.

### CODE REVIEW

File: srcpkgs/freecad/patches/002-execinfo.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
++#ifdef __GLIBC__
+     void *callstack[128];
+     size_t nMaxFrames = sizeof(callstack) / sizeof(callstack[0]);
+     size_t nFrames = backtrace(callstack, nMaxFrames);
+@@ -1105,6 +1108,8 @@
+     }
+ 
+     free(symbols);
++#endif
++    std::cerr << "sorry, no backtrace on musl libc";
```
Comment (Gottox) [inline srcpkgs/freecad/patches/002-execinfo.patch]: this always returns the error message, even on glibc.

File: srcpkgs/freecad/patches/010-salomesmesh-execinfo.patch [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
++#ifdef __GLIBC__
+     void *array[50];
+     size_t size;
+     char **strings;
+@@ -123,6 +126,8 @@
+       }
+ 
+     free (strings);
++#endif
++   std::cerr << "sorry, no backtrace on musl libc";
```
Comment (Gottox) [inline srcpkgs/freecad/patches/010-salomesmesh-execinfo.patch]: Same here

**Comment by newbluemoon** [comment]
For musl there’s a libexecinfo package. You could try if that worked and then the patches wouldn’t be necessary.

**Comment by pullmoll** [comment]
FYI fixed med by creating a hdf5-18 package.

**Comment by Piraty** [comment]
thanks @pullmoll .
The patches proposed in this PR solve build issue on musl for now. I didn't succeed in making the cmake/linker use libexeinfo correctly, so backtrace won't be available.

**Pull request closed without merging**